### PR TITLE
feat(EMI-1778): list privacy popover 

### PR DESCRIPTION
**Pull request by rquartararo**
This PR resolves [EMI-1778] and [EMI-1786] 

### Description

This implements the list privacy eye icon popover and fixes a copy issue. 

https://github.com/artsy/eigen/assets/50849237/75124963-f62f-4d8b-97d3-01ecb1a0b005

### PR Checklist

- [x] I have tested my changes on **iOS** and **Android**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [x] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- artwork list privacy popover 

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[EMI-1778]: https://artsyproduct.atlassian.net/browse/EMI-1778?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[EMI-1786]: https://artsyproduct.atlassian.net/browse/EMI-1786?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
### This PR contains the following changes:

- Cross-platform user-facing changes (artwork list privacy popover)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 60144ad90bfd7e30333c9526f9e78aa82e95192f
</p>

**Review by fladson — APPROVED** [review]

File: src/app/Components/ArtworkLists/views/ArtworkListOfferSettingsView/components/ArtworkListOfferSettingsHeader.tsx [hunk 1]
```diff
@@ -14,7 +14,7 @@ export const ArtworkListOfferSettingsHeader = () => {
 
       <Text variant="xs">
         Shared lists are eligible to receive offers from galleries. Switching sharing off will make
-        them visible only to you, and you won't receive offers{" "}
+        them visible only to you, and you won't receive offers.{" "}
```
Comment (starsirius) [inline src/app/Components/ArtworkLists/views/ArtworkListOfferSettingsView/components/ArtworkListOfferSettingsHeader.tsx:17]: Thank you for taking care of the copy! Can you also help update [the first bullet point](https://artsyproduct.atlassian.net/browse/EMI-1786) of the ticket which _I think_ maybe it's [here](https://github.com/artsy/eigen/blob/653f4cb0eff2d61be58117813fbe24a364c5958c/src/app/Scenes/ArtworkLists/SavesTabHeader.tsx#L69).

Comment (starsirius) [reply src/app/Components/ArtworkLists/views/ArtworkListOfferSettingsView/components/ArtworkListOfferSettingsHeader.tsx:17]: Ah realized my link is outdated! 😬  Thank you for double checking!

**Review by starsirius — APPROVED** [review]
🎉  Started a discussion to make sure [design is aligned](https://artsy.slack.com/archives/C02JHHHKP5K/p1712757243219189?thread_ts=1712754321.055179&cid=C02JHHHKP5K) with the black translucent overlay. Otherwise looks great to me!

**Pull request merged**